Qingc

Mortick opened the pull request.

Commit:
Updates

Commit:
Updates

Commit:
Updates

Commit:
$entry to $entity fix for saving metadata

Commit:
Kaltura server video deletion

Commit:
null for $type values in field_kaltura_build_embed is changed to KALTURA_MEDIA_TYPE_VIDEO.

Fixed try and catch exception ex in kaltura_derivatives.inc.

Commit:
null for $type values in field_kaltura_build_embed is changed to KALTURA_MEDIA_TYPE_VIDEO.

Fixed try and catch exception ex in kaltura_derivatives.inc.

Commit:
null for $type values in field_kaltura_build_embed is changed to KALTURA_MEDIA_TYPE_VIDEO.

Fixed try and catch exception ex in kaltura_derivatives.inc.

ctgraham commented on includes/kaltura_derivatives.inc:
```diff
@@ -5,12 +5,36 @@
-  if ($force || !isset($object['KALTURA_STREAM'])) {
+    //If the object's KALTURA_STREAM is null then we add it into the file system
+    //using islandora_video_copy_obj_datastream
+    if ($force || !isset($object['KALTURA_STREAM'])) {
```
Watch for unintended whitespace changes.  Here it appears that `if` is indented unexpectedly.  Match your coding style against the existing coding style.  For example, here indentation has been done with two spaces in the rest of the module.

ctgraham commented on includes/kaltura_derivatives.inc:
```diff
@@ -5,12 +5,36 @@
+            //decode $old_content for entry_id
+            $decoded = json_decode($old_content, true);
+            //find entity through $decoded string 
+            $entity = kaltura_entry_load($decoded['entry_id']);
```
What if the KALTURA_STREAM exists, but is a bad or broken format (invalid JSON)?  You don't have any error checking on fetching `$decoded`, `$entity`, or `$old_entity_id`.

ctgraham commented on includes/kaltura_derivatives.inc:
```diff
@@ -5,12 +5,36 @@
+        }
+    }
+    
+    $prevJSON =$object['KALTURA_STREAM'];
```
`$prevJSON` looks like dead code.

ctgraham commented on includes/kaltura_derivatives.inc:
```diff
@@ -26,28 +50,44 @@ function islandora_video_streaming_kaltura_upload($object, $force = false) {
-      // TODO: duplicate metadata via kaltura_save_entry_metadata() ?
-      kaltura_entry_save($entity);
+            
+      kaltura_save_entry_metadata($entity);
```
Islandora supplies us with rich metadata, likely in the MODS or DC datastream.  I don't know if we are guaranteed a MODS or DC at the time of processing this upload, though.  What kinds of metadata (title, format, length, etc.?) does `kaltura_entry_save` expect to be recorded in the `$entity`?  If this is accessible, we should try to add it.

Mortick replied:
_What kinds of metadata (title, format, length, etc.?) does kaltura_entry_save expect to be recorded in the $entity?_  Unknown

Mortick replied:
_What kinds of metadata (title, format, length, etc.?) does kaltura_entry_save expect to be recorded in the $entity?_ Unknown, the function kaltura_entry_save uses function entity_save.

ctgraham replied:
Sorry, should have referenced `kaltura_save_entry_metadata()` here, not `kaltura_entry_save`.  Take a look at how metadata is leveraged in the Kaltura Media Entries in Drupal.  If there is metadata about the conceptual object (title, description) or about the digital format (length, format), we can potentially add it here.  Otherwise, we may not even need to call `kaltura_save_entry_metadata()`.

Mortick replied:
The metadata about the Kaltura Media Entries only contain title, descriptions, and tags. Also pushed changes needed + some Panopto code.

ctgraham replied:
Every Islandora Object has a formal title and description, as well as many subject and keyword elements which might be considered Kaltura "tags".  I think our goal with the Kaltura metadata should be twofold: 1) to apply a human readable element to the video, and 2) to backlink into islandora.  For that purpose, let's copy the Islandora Object's "Label" into the Kaltura Description, and the Islandora Object's PID (Persistent Identifier) into the Kaltura Title.  This should, in a Kaltura interface, present the Videos by Islandora Object, but also provide a human-readable Description.

ctgraham commented on includes/kaltura_derivatives.inc:
```diff
@@ -56,7 +96,14 @@ function islandora_video_streaming_kaltura_upload($object, $force = false) {
       }
-      return islandora_video_add_datastream($object, 'KALTURA_STREAM', $temp_name);
+      
+      $arr = islandora_video_add_datastream($object, 'KALTURA_STREAM', $temp_name);
```
This is meaningless variable naming.  Perhaps `$result` or `$resultMessage` or similar?

ctgraham commented on islandora_video_streaming.module:
```diff
@@ -62,6 +62,7 @@ function islandora_video_streaming_get_available_viewers() {
       return $b['module_weight'] - $a['module_weight'];
     }
   );
+ 
```
Watch for unintended whitespace changes.  This will clutter the commit and PR.

ctgraham commented on islandora_video_streaming.module:
```diff
@@ -85,3 +86,6 @@ function islandora_video_streaming_islandora_video_admin_submit($form, &$form_st
   }
 }
 
+/*
```
This is also the equivalent of an unwanted whitespace change.

ctgraham reviewed (changes requested):
Good initial changeset.  Mostly code style changes needed, with a couple of notes on functional questions/concerns:
* a bit of extra error handling required in the deletion of the existing Drupal data
* additional consideration needed with regard to what metadata to push to Kaltura (or if it is needed).

Commit:
Format issues have been fixed. 

Dead code erased and checks are added for for $decoded, $entity, and $old_entity_id.

$arr is changed to $result

Commit:
Revert "Format issues have been fixed. "

This reverts commit e591bc87804a5d6c9d93d1e39f9d5bbfdbd11a57.

Commit:
Removed added Panopto code
